test/extensions/filters/http/common/fuzz/uber_filter.h

```diff
@@ -60,6 +64,18 @@ class UberFilterFuzzer {
     ON_CALL(factory_context_.admin_, removeHandler(_)).WillByDefault(testing::Return(true));
   }
 
+  void addProtoDescriptor(absl::string_view filter_name, Protobuf::Message* message) {
+    if (filter_name.find("grpc_json_transcoder") != absl::string_view::npos) {
+      envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder& config =
+          dynamic_cast<
+              envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder&>(
+              *message);
+      config.add_services("bookstore.Bookstore");
+      config.set_proto_descriptor(TestEnvironment::runfilesPath("test/proto/bookstore.descriptor"));
```

Comment:
If the message is linked in to the binary, then I think you can call `message.GetDescriptor()` to get the descriptor.  Would that allow you to avoid reading the file?

Docs on `Message::GetDescriptor()`:
https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.message#Message.GetDescriptor

Reply:
Oooh. That's useful, thank you! This is a file descriptor for the whole `bookstore.proto` -- But I think there must be something analogous for FileDescriptor too. Will look into.

Reply:
@asraa yep, you probably want `Protobuf::DescriptorPool::generated_pool()->FindFileByName` or something like this. Then you can serialize that to a binary string and use `set_proto_descriptor_bin`

Reply:
It worked!!!! Thank you both so much, avoided the filesystem mess. Still a little ugly. I'm hoping to find a way to automagically find the deps I need in the `FileDescriptorSet`. (I realized `bookstore.descriptor` was compiled with some additional deps, and without them, JsonTranscoderConfig will be unable to build the descriptor set.)